### PR TITLE
Add custom volume and volume for secrets

### PR DESCRIPTION
**Pull request by leszko**
Add two volumes (disabled by default):
- Secret Volume (for the use case with storing `keystore`/`truststore` as secrets)
- Custom Volume (for any user-defined volume (e.g. to mount custom JARs stored as `ConfigMap`)

### CODE REVIEW

File: stable/hazelcast-enterprise/README.md [hunk 1]
```diff
@@ -81,6 +81,8 @@ The following table lists the configurable parameters of the Hazelcast chart and
 | `securityContext.enabled`                  | Enables Security Context for Hazelcast and Management Center                                                   | `true`                                               |
 | `securityContext.runAsUser`                | User ID used to run the Hazelcast and Management Center containers                                             | `1001`                                               |
 | `securityContext.fsGroup`                  | Group ID associated with the Hazelcast and Management Center container                                         | `1001`                                               |
+| `secretsMountName`                         | Secret name that is mounted as '/data/secrets/' (e.g. with keystore/trustore files)                            | `nil`                                                |
+| `customVolume`                             | Configuration for a volume mounted as '/data/custom' (e.g. to mount a volume with custom JARs)                 | `nil`                                                |
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/README.md]: Can we briefly explain how those are created (secrets and volumes) in README? We can give some crosslink as well. Not too detailed but something for people to start with.

File: stable/hazelcast-enterprise/templates/statefulset.yaml [hunk 2]
```diff
@@ -92,3 +100,12 @@ spec:
...
         configMap:
           name: {{ template "hazelcast.fullname" . }}-configuration
+      {{- if .Values.secretsMountName }}
+      - name: hazelcast-secrets
+        secret:
+          secretName: {{ .Values.secretsMountName }}
+      {{- end }}
+      {{- if .Values.customVolume }}
+      - name: hazelcast-custom
+{{ toYaml .Values.customVolume | indent 8 }}
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/statefulset.yaml]: What does this do as It has different convention than secretName?

`toYaml .Values.customVolume | indent 8 `

Comment (leszko) [reply stable/hazelcast-enterprise/templates/statefulset.yaml]: Hmm, it's super generic. It literally pastes the content of the `.Values.customVolume`, so you can specify any volume you like. My way of thinking is that: 

- `secretsMountName` - you use it for `keystore`/`truststore`, which is always strored as a `Secret` in k8. That is the recommended way.
- `customVolume` - can be any volume type defined by the user. The use case here is actually the custom JARs. And people may want to store them in `ConfigMap` or in `Secret`, but if they're big then people may like to store them in [any of the volume type](https://kubernetes.io/docs/concepts/storage/volumes/#types-of-volumes), for example, in Azure disk or GCP Perisistent Disk. That is why we need to be super generic here and the user provides all the configuration.

File: stable/hazelcast-enterprise/README.md (hunk 1 above)
Comment (leszko) [reply stable/hazelcast-enterprise/README.md]: Added a section "Configuring SSL".

**Pull request merged**